src/Adapters/Etw.Net451/ITraceEventSession.cs

```diff
@@ -0,0 +1,81 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ITraceEventSession.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.ApplicationInsights.EtwCollector
+{
+    using System;
+    using Diagnostics.Tracing.Session;
+    using Microsoft.Diagnostics.Tracing;
+
+    /// <summary>
+    /// Abstracts properties and methods of <see cref="Microsoft.Diagnostics.Tracing.Session.TraceEventSession"/> to provide better test ability.
+    /// </summary>
+    internal interface ITraceEventSession : IDisposable
+    {
+        /// <summary>
+        /// If this is a real time session you can fetch the source associated with the session
+        /// to start receiving events. Currently does not work on file based sources (we
+        /// expect you to wait until the file is complete).
+        /// </summary>
+        ETWTraceEventSource Source { get; }
+
+        /// <summary>
+        /// Is the current process Elevated (allowed to turn on a ETW provider). This is
+        /// useful because you need to be elevated to enable providers on a TraceEventSession.
+        /// </summary>
+        bool? IsElevated();
+
+        /// <summary>
+        /// Enable a NON-KERNEL provider (see also EnableKernelProvider) which has a given
```

Comment:
Don't copy docs from other libraries without reading them first 😃 

Since you do not have `EnableKernelProvider`, this is just too verbose and confusing. Just say "Enables ETW provider with given provider name" or something like that.